### ISSUE TITLE
"Apply color scheme to all items" can irreversably dye items that "can not be dyed"

### ISSUE THREAD

**Issue by 5119ror** [WarEmu/WarBugs#17789]
**Expected behavior and actual behavior:**
Expected: if an item "can not be dyed", then it should never accept dye color
(Bonus expected behavior: "Scouring Agent" should remove any applied dye color from an item even if that item "can not be dyed")

Actual: the "Apply color scheme to all items" checkbox at the vendor "dye items" UI will in fact dye items that "can not be dyed" — and, unfortunately, any dye color that is applied in this way can not be removed by "Scouring Agent" because of the "can not be dyed" issue

Sov cloak (with alternate appearance) dyed via vendor:
![image](https://user-images.githubusercontent.com/69434505/93368843-5ef61f80-f803-11ea-8fd2-311f83c27f38.png)
![image](https://user-images.githubusercontent.com/69434505/93368980-88af4680-f803-11ea-9e8c-073c81046de8.png)

Attempting to use Scouring Agent:
![image](https://user-images.githubusercontent.com/69434505/93368700-29e9cd00-f803-11ea-8de8-e9bd6ef78c60.png)

**Comment by 5119ror** [comment]
Oh, this appears to be a dupe of #16193 please feel free to close
I'm not sure I understand the workaround mentioned in the comments of that issue though — I can't seem to actually remove the dye?

**Comment by MaxHayman** [comment]
If you remove all gear apart from this can you dye it again?

**Comment by 5119ror** [comment]
Yes, by removing all gear other than this item, you can re-dye it to any color that the vendor "dye items" UI offers (this is actually what I am doing in the screenshot above).

Because the "scouring agent" item seems to follow item-based dye rules ("item-based" as in: a bottle of dye in your inventory), "scouring agent" can't be used on items that "can not be dyed". And the vendor "dye items" UI does not offer any kind of "scouring agent"-type functionality. This all unfortunately means (as far as I can tell) that if you accidentally dye a "can not be dyed" item in this way, there is no way to revert that item to its original appearance.

**Comment by 5119ror** [comment]
I wonder if it would be possible for the scouring agent check to be changed from "if item can be dyed" to "if item has dyes applied" and then if that check passes, do the scouring agent routine of removing all dyes.

**Comment by Goshban** [comment]
Same problem with the sovereign belt of shaman i can't renove the secondary color by the scouring agent
